### ISSUE TITLE
Publish most recent version of the code to npm as `head` tag

### ISSUE THREAD

**Issue by alexander-fenster** [googleapis/repo-automation-bots#4078]
This is coming from https://github.com/google/gts/issues/554 and looks like a great idea for automation. What if we publish the most recent library code to npm and tag it as `head` before making any release?

**Comment by chingor13** [comment]
Are you thinking something like on release PR creation and synchronization (a push the release PR branch), trigger a separate release job for publishing a snapshot/preview binary?

The actual snapshot/preview release script would need to be handled by each language team.

We could possibly build some kind of trigger like this for the next-gen release conductor (SLSA compliant release pipeline app), but we probably won't add it to the current, legacy release triggering mechanism.

**Comment by alexander-fenster** [comment]
Yeah, I'm guessing, whenever the release PR gets updated (the new item gets added to the changelog), some code might publish the current `main` with `--tag head` (I'm talking about NPM now, not sure if all package managers have a way to publish non-default version). It will make it much easier to test stuff before making actual release (just `npm install @google-cloud/speech@head`). No rush at all with this feature, just something to be considered for the future.
